BUILD.md

```diff
@@ -2,57 +2,64 @@
 
 ## Prepare your local environment
 
-The build scripts in this repository require bash 4+. To update to newer bash on OS X, see:
-https://johndjameson.com/blog/updating-your-shell-with-homebrew/
+The build scripts in this repository require:
 
-## Adding packages to the base image
+- bash 4+. To update to newer bash on OS X, see: https://johndjameson.com/blog/updating-your-shell-with-homebrew/
+- Docker Desktop. To build multi-arch images (heroku-24 and beyond), enable
+  the `containerd` snapshotter feature should be enabled.
 
-Add the package you want to the appropriate `setup.sh` for example `heroku-22/setup.sh`:
+## Build
 
-```diff
-+    libc6-dev \
-```
+To build the base images locally, run this from the repo root:
 
-Once done, run the `bin/build.sh` locally to generate the corresponding `installed-packages.txt`.
+    bin/build.sh STACK_VERSION
 
-The `*-build` variants include all the packages from the non-build variant by default. This means that if you're adding a package to both, you only need to add them to the non-build variant. The example above will add `libc6-dev` to both `heroku-22` and `heroku-22-build`.
+For example:
 
-The `*cnb*` variants inherit the installed packages from the non-`*cnb*` variant. Add packages to a non-`*cnb*` variant to add them to the `*cnb*` variant.
+    ./bin/build.sh 24
 
-## Build
+### Multi-architecture images (heroku-24 and later)
 
-To build the base images locally, run this from the repo root:
+This script will build a family of 2 images:
 
-    bin/build.sh STACK_VERSION
+* `heroku/heroku:{STACK_VERSION}` - A multi-architecture manifest list of Heroku base runtime images supporting `amd64` and `arm64` architectures
+* `heroku/heroku:{STACK_VERSION}-build` - A multi-architecture manifest list of Heroku base build images supporting `amd64` and `arm64` architectures
 
-For example:
+### Single architecture images (heroku-22 and prior)
+
+This script will build a family of 4 images:
 
-    ./bin/build.sh 22
+* `heroku/heroku:{STACK_VERSION}` - The Heroku base run image supporting `amd64` architecture
+* `heroku/heroku:{STACK_VERSION}-build` - The Heroku base build image supporting `amd64` architecture
+* `heroku/heroku:{STACK_VERSION}-cnb` - The Heroku base run image for Cloud Native Buildpacks supporting `amd64` architecture
+* `heroku/heroku:{STACK_VERSION}-cnb-build` - The Heroku base build image for Cloud Native Buildpacks supporting `amd64` architecture
+
+## Adding packages to the base image
 
-If you're building on a machine with an architecture other than amd64, set `DOCKER_DEFAULT_PLATFORM` to the appropriate "`linux/amd64`" value in the environment:
+Add the package you want to the appropriate `setup.sh` for example `heroku-24/setup.sh`:
+
+```diff
++    libc6-dev \
```

Comment:
(No line continuation is needed any more, now that a bash ~array like pattern is used)

```suggestion
+    libc6-dev
```